src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -326,90 +358,67 @@ protected function guardAgainstInvalidTranslationResetRequest($content)
     }
 
     /**
-     * @param $lang
-     * @param $type
-     * @param string $theme Selected theme name
-     * @param null $search
+     * Returns a translation domain tree
      *
-     * @return array
-     */
-    private function getNormalTree($lang, $type, $theme, $search = null)
-    {
-        $treeBuilder = new TreeBuilder($this->translationService->langToLocale($lang), $theme);
-        $catalogue = $this->translationService->getTranslationsCatalogue($lang, $type, $theme, $search);
-
-        return $this->getCleanTree($treeBuilder, $catalogue, $theme, $search);
-    }
-
-    /**
-     * @param string $lang Two-letter iso code
-     * @param string $selectedModuleName Selected module name
-     * @param string|null $search
+     * @param string $lang
+     * @param string $type "themes", "modules", "mails", "mails_body", "back" or "others"
+     * @param array $search Search strings
+     * @param string|null $theme Selected theme name. Set only if type = "themes"
+     * @param string|null $module
      *
      * @return array
-     */
-    private function getModulesTree($lang, $selectedModuleName, $search = null)
-    {
-        $theme = null;
-        $locale = $this->translationService->langToLocale($lang);
-
-        $moduleProvider = $this->container->get('prestashop.translation.external_module_provider');
-        $moduleProvider->setModuleName($selectedModuleName);
-
-        $treeBuilder = new TreeBuilder($locale, $theme);
-        $catalogue = $treeBuilder->makeTranslationArray($moduleProvider, $search);
-
-        return $this->getCleanTree($treeBuilder, $catalogue, $theme, $search, $selectedModuleName);
-    }
-
-    /**
-     * @param string $lang Two-letter iso code
-     * @param null $search
      *
-     * @return array
+     * @throws Exception
      */
-    private function getMailsSubjectTree($lang, $search = null)
+    private function getTree(string $lang, string $type, array $search, ?string $theme = null, ?string $module = null)
     {
-        $theme = null;
+        $locale = $this->translationService->langToLocale($lang);
 
-        $treeBuilder = new TreeBuilder($this->translationService->langToLocale($lang), $theme);
-        $catalogue = $this->translationService->getTranslationsCatalogue($lang, 'mails', $theme, $search);
+        $catalogue = $this->translationService->getTranslationsCatalogue(
+            $this->buildProviderType($type, $theme, $module),
```

Comment:
Same here, comments must be added to explain why $theme and $module are mandatory to build a $type

Reply:
up

Reply:
The reason is the same as https://github.com/PrestaShop/PrestaShop/pull/19487#discussion_r537405476
Depending on the type, the selectedValue has different meaning. I can use a single if you're more comfortable with like taht